[FIX] -k option should keep apps, dataValues, documents, etc. upon restart

cgbautista opened the pull request:
Moved 4 functions inside a condition to avoid resetting container folders to the original content (it didn't erase contents, but overwrote any change on those folders with the files' previous version)

cgbautista committed:
Moved 4 functions inside a condition to avoid reseting container folders
to the original content (it didn't erase contents, but overwrote any
change on those folders with the files' previous version)

cgbautista committed:
extracted setup_tomcat, since it will copy any dhis.conf or server.xml
that is passed as arguments

tokland commented on src/d2_docker/config/dhis2-core-start.sh:
```diff
@@ -163,13 +163,13 @@ run() {
-        debug "Container: already configured. Skip DB load"
+        debug "Container: already configured. Skip DB load and keeping other changes"
     else
+	debug "Container: clean. Copying tomcat files and dhis folders"
```
I think there is a tab/spaces mismatch there

tokland reviewed (approved):
Tested with cpr instances: start without -k, stop, start with -k, start. all good

cgbautista committed:
Fix in tabs

cgbautista committed:
After stopping the running containers, the command "docker compose ps
-q" did not return any container, so a "-a" parameter is added to ensure
all processes (even stopped) are returned

ifoche committed:
Merge pull request #153 from EyeSeeTea/fix/rm_command_did_not_clear_stopped_containers

"d2-docker rm" not clearing up containers

ifoche reviewed (approved):
thanks @cgbautista
